src/extensions/extensions.injectable.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { Injectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { computed, IComputedValue } from "mobx";
+import { ExtensionLoader } from "./extension-loader";
+import type { LensExtension } from "./lens-extension";
+
+const extensionsInjectable: Injectable<
+  IComputedValue<LensExtension[]>,
+  { extensionLoader: ExtensionLoader }
+> = {
+  getDependencies: () => ({
+    extensionLoader: ExtensionLoader.createInstance(),
```

Comment:
No you aren't supposed to create the instances in more than once place.

Reply:
Yep, this is the issue with current `Singleton`. I shouldn’t need to know how and when instance is created. 

If we change this to `.getInstance()` we need to worry about that `.createInstance()` has been called. 

This is something that we need to fix by getting rid of `Singleton`

Reply:
I completely disagree, just because something is a singleton doesn't mean there aren't dependency's on other instances, and those instances might need arguments to be created.

Reply:
Solution is to use DI to make instances. That’s how I don’t need to worry about creating instance and getting arguments somewhere. 

In this case, it’s clear that `ExtensionLoader` needs to be injected. Therefore `injectable` should be created.

Reply:
I think @Nokel81 means that we shouldn't call `ExtensionLoader.createInstance()` in multiple places (see main/index.ts for other call). That said, it's a bit unfortunate that caller needs to know anything about this -> maybe instance should always come via injectable.

Reply:
I know and I agree. 

To be honest, I’m not sure why it was left as `.createInstance()` since `.getInstance()` is more than enough. 

However it works as a great example of an issue caused `Singleton` and permits this kind discussion. 

I’ll change it tomorrow anyway :)

Reply:
Because not every place that consumes an instance needs/should know how to create it.

Reply:
This we know ;)